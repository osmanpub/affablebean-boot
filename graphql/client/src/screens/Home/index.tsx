import React from "react";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Categories as CategoriesState } from "../../interfaces/categories";
import { Match } from "../../interfaces/router";

type Props = {
  categories: CategoriesState;
  match: Match;
};

function Home(props: Props) {
  const { match } = props;
  const url = match ? match.url : "";

  return (
    <div>
      <Header url={url} />
      {<Categories />}
      <Footer />
    </div>
  );
}

export default Home;
