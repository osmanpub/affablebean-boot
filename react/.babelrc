 
{
  "presets": ["@babel/preset-react"],
  "env": {
    "test": {
      "plugins": [
        ["istanbul", {
          "exclude": [
            "**/*.spec.js"
          ]
        }]
      ]
    }
  },
  "plugins": ["transform-class-properties", "istanbul"]
}
