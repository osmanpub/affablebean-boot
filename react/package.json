{
  "name": "affablebean",
  "version": "0.1.0",
  "private": true,
  "eslintConfig": {
    "extends": "react-app",
    "plugins": [
      "react-hooks"
    ],
    "rules": {
      "react-hooks/rules-of-hooks": "error",
      "react-hooks/exhaustive-deps": "warn"
    }
  },
  "scripts": {
    "build": "react-scripts build",
    "coveralls": "cat coverage/lcov.info | coveralls",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run",
    "cypress:verify": "cypress verify",
    "eject": "react-scripts eject",
    "report:coverage": "nyc report --reporter=html",
    "report:coverage:text": "nyc report --reporter=text",
    "start": "node_modules/react-scripts/bin/react-scripts.js start &",
    "start:java": "cd .. && sh server && cd react && node_modules/react-scripts/bin/react-scripts.js start &",
    "test": "react-scripts test"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@reduxjs/toolkit": "^1.2.5",
    "axios": "^0.19.2",
    "node-rest-client": "^3.1.0",
    "react": "^16.12.0",
    "react-alert": "^6.0.0",
    "react-alert-template-basic": "^1.0.0",
    "react-dom": "^16.12.0",
    "react-hook-form": "^4.9.7",
    "react-redux": "^7.2.0",
    "react-router-dom": "^5.1.2",
    "react-scripts": "3.4.0",
    "redux": "^4.0.5",
    "styled-components": "^5.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/preset-react": "^7.8.3",
    "@cypress/code-coverage": "^1.12.2",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.4.0",
    "@testing-library/user-event": "^7.2.1",
    "@types/axios": "^0.14.0",
    "@types/cypress": "^1.1.3",
    "@types/jest": "^25.1.3",
    "@types/react": "^16.9.21",
    "@types/react-alert": "^5.2.0",
    "@types/react-bootstrap": "^0.32.20",
    "@types/react-dom": "^16.9.5",
    "@types/react-redux": "^7.1.7",
    "@types/react-router-dom": "^5.1.3",
    "@types/react-test-renderer": "^16.9.2",
    "@types/redux": "^3.6.0",
    "@types/redux-mock-store": "^1.0.2",
    "@types/styled-components": "^5.0.0",
    "babel-plugin-istanbul": "^6.0.0",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "coveralls": "^3.0.9",
    "cross-env": "^7.0.0",
    "cypress": "^4.0.2",
    "istanbul-lib-coverage": "^3.0.0",
    "nyc": "^15.0.0",
    "react-test-renderer": "^16.12.0",
    "redux-mock-store": "^1.5.4",
    "typescript": "^3.8.2"
  },
  "nyc": {
    "sourceMap": false,
    "instrument": false
  }
}
