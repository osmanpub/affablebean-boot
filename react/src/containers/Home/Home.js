import React, { Component } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { fetchCategoriesIfNeeded } from "../../net/categories";

export class Home extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchCategoriesIfNeeded());
  }

  render() {
    const { cart, categories, match } = this.props;
    const { items } = categories;

    if (items.length === 0) {
      return null;
    }

    return (
      <div>
        <span id="hello">hello</span>
        <Header cart={cart} url={match.url} />
        <Categories categories={items} />
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { cart, categories } = state;

  return {
    cart,
    categories
  };
};

export default connect(mapStateToProps)(Home);
