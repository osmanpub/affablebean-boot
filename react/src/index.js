import React from "react";
import { render } from "react-dom";
import { configureStore } from "redux-starter-kit";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import reducer from "./reducers";
import App from "./containers/App";
import CategoryProducts from "./containers/CategoryProducts";
import * as serviceWorker from "./serviceWorker";

const store = configureStore({
  reducer
});

render(
  // <React.StrictMode>
  <Provider store={store}>
    <Router>
      <Route path="/" exact component={App} />
      <Route path="/category/:id" component={CategoryProducts} />
    </Router>
  </Provider>,
  // </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
