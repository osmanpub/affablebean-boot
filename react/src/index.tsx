import { configureStore } from "@reduxjs/toolkit";
import React, { lazy, Suspense } from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import ErrorBoundary from "./components/ErrorBoundary";
import rootReducer from "./redux";
import * as serviceWorker from "./serviceWorker";
const Cart = lazy(() => import("./containers/Cart"));
const CategoryProducts = lazy(() => import("./containers/CategoryProducts"));
const Checkout = lazy(() => import("./containers/Checkout"));
const Contact = lazy(() => import("./containers/Contact"));
const Home = lazy(() => import("./containers/Home"));
const Privacy = lazy(() => import("./containers/Privacy"));

const store = configureStore({
  reducer: rootReducer
});

render(
  <React.StrictMode>
    <ErrorBoundary>
      <Provider store={store}>
        <Router>
          <Suspense fallback={<div>Loading...</div>}>
            <Switch>
              <Route path="/" exact component={Home} />
              <Route path="/category/:id" component={CategoryProducts} />
              <Route path="/checkout" component={Checkout} />
              <Route path="/contact" component={Contact} />
              <Route path="/privacy" component={Privacy} />
              <Route path="/viewCart/:clear" component={Cart} />
            </Switch>
          </Suspense>
        </Router>
      </Provider>
    </ErrorBoundary>
  </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
