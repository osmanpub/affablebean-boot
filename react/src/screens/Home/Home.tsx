import React, { useEffect } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Props } from "../../interfaces/props";
import { fetchCategoriesIfNeeded } from "../../net/categories";
import { RootState } from "../../redux";

export function Home(props: Props) {
  useEffect(() => {
    const { dispatch } = props;
    dispatch(fetchCategoriesIfNeeded());
  }, []);

  const { cart, categories, match } = props;
  const { items } = categories;
  const url = match ? match.url : "";

  if (items.length === 0) {
    return null;
  }

  return (
    <div>
      <Header cart={cart} url={url} />
      <Categories categories={items} />
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
  categories: state.categories
});

export default connect(mapStateToProps)(Home);
