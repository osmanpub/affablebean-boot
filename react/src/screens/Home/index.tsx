import React, { useEffect } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Categories as CategoriesState } from "../../interfaces/categories";
import { fetchCategoriesIfNeeded } from "../../net/categories";
import { RootState } from "../../redux";

type Props = {
  categories: CategoriesState;
  dispatch: Function;
  match: any;
};

function Home(props: Props) {
  const { categories, dispatch, match } = props;
  const { items } = categories;
  const url = match ? match.url : "";

  useEffect(() => {
    dispatch(fetchCategoriesIfNeeded());
  }, [dispatch]);

  if (items.length === 0) {
    return null;
  }

  return (
    <div>
      <Header url={url} />
      <Categories categories={items} />
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
  categories: state.categories
});

export default connect(mapStateToProps)(Home);
