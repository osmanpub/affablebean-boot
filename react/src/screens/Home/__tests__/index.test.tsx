import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Home from "..";

const mockStore = configureStore([thunk]);

const initialState = {
  categories: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        id: 1,
        name: "dairy"
      },
      {
        id: 2,
        name: "meats"
      },
      {
        id: 3,
        name: "bakery"
      },
      {
        id: 4,
        name: "fruit & veg"
      },
      {
        id: 5,
        name: "cereals"
      },
      {
        id: 6,
        name: "drinks"
      }
    ]
  }
};

const store = mockStore(initialState);

const homeScreen = (
  <Router>
    <Provider store={store}>
      <Home match={{ path: "/", url: "/", isExact: true, params: {} }} />
    </Provider>
  </Router>
);

describe("<Home />", () => {
  it("renders correctly", () => {
    const component = renderer.create(homeScreen).toJSON();
    // @ts-ignore
    expect(component).toMatchSnapshot();
  });

  it("show smaller navigatioal logo, on left", () => {
    const { getByAltText } = render(homeScreen);
    const intro = getByAltText(/Affable Bean logo/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });

  it("show larger logo, on right", () => {
    const { getByAltText } = render(homeScreen);
    const intro = getByAltText(/the affable bean/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });

  it("don't show view cart link", () => {
    const { getByText } = render(homeScreen);
    const link = getByText(/view cart/i);
    // @ts-ignore
    expect(link).not.toBeVisible();
  });

  it("don't show proceed to checkout link", () => {
    const { getByText } = render(homeScreen);
    const link = getByText(/proceed to checkout/i);
    // @ts-ignore
    expect(link).not.toBeVisible();
  });

  it("don't show cart header total", () => {
    const { getByTestId } = render(homeScreen);
    const cart = getByTestId(/cart-hdr-total/i);
    // @ts-ignore
    expect(cart).not.toBeVisible();
  });

  it("show welcome paragraph", () => {
    const { getByTestId } = render(homeScreen);
    const paragraph = getByTestId(/home-welcome/i);
    // @ts-ignore
    expect(paragraph).toBeInTheDocument();
  });

  it("show six categories", () => {
    const { getAllByTestId } = render(homeScreen);
    const boxes = getAllByTestId(/categoryBox/i);
    // @ts-ignore
    expect(boxes).toHaveLength(6);
  });

  it("show privacy", () => {
    const { getByTestId } = render(homeScreen);
    const intro = getByTestId(/privacy/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });

  it("show contact", () => {
    const { getByTestId } = render(homeScreen);
    const intro = getByTestId(/contact/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });
});
