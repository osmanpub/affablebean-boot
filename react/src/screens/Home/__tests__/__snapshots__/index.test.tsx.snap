// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<div>
  <div
    className="sc-fzozJi gYhGck"
  >
    <div
      className="sc-fzoLsD kcEwNV"
    >
      <div
        className="sc-AxheI dnzAqr"
      >
        <div
          className="sc-Axmtr ivQZfu headerWidget"
        >
          <a
            className="bubble"
            data-cy="proceed-checkout"
            href="/checkout"
            onClick={[Function]}
          >
            proceed to checkout ➟
          </a>
        </div>
        <div
          className="sc-AxmLO dmEyYL headerWidget"
        >
          <span
            className="horizontalMargin"
            data-cy="cart-hdr-total"
            data-testid="cart-hdr-total"
          >
            <img
              alt="shopping cart icon"
              src="/static/img/cart.gif"
            />
             
          </span>
          <a
            className="bubble"
            data-cy="cart-hdr-view-cart"
            href="/viewCart/false"
            onClick={[Function]}
          >
            view cart
          </a>
        </div>
      </div>
      <div
        className="sc-fzpans iWSqNc"
      >
        <a
          data-cy="home-logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Affable Bean logo"
            src="/static/img/logo.png"
          />
        </a>
      </div>
      <img
        alt="the affable bean"
        id="logoText"
        src="/static/img/logoText.png"
      />
    </div>
  </div>
  <div>
    <div
      className="sc-AxirZ jHmcxa"
    >
      <div
        className="sc-AxhCb cVYmNY"
      >
        <p
          className="sc-AxjAm dUXPTK"
          data-cy="home-welcome"
          data-testid="home-welcome"
        >
          Welcome to the online home of the Affable Bean Green Grocer.
        </p>
        <p>
          Our unique home delivery service brings you fresh organic produce, dairy, meats, breads and other delicious and healthy items direct to your doorstep.
        </p>
      </div>
    </div>
    <div
      className="sc-AxiKw kHzIbN"
    >
      <span
        className="categoryBox"
        data-testid="categoryBox"
      >
        <a
          href="/category/1"
          onClick={[Function]}
        >
          <span
            className="categoryLabel"
          />
          <span
            className="categoryLabelText"
          >
            dairy
          </span>
          <img
            alt="dairy"
            className="categoryImage"
            data-cy="category-dairy"
            data-testid="category-dairy"
            src="/static/img/categories/dairy.jpg"
          />
        </a>
      </span>
      <span
        className="categoryBox"
        data-testid="categoryBox"
      >
        <a
          href="/category/2"
          onClick={[Function]}
        >
          <span
            className="categoryLabel"
          />
          <span
            className="categoryLabelText"
          >
            meats
          </span>
          <img
            alt="meats"
            className="categoryImage"
            data-cy="category-meats"
            data-testid="category-meats"
            src="/static/img/categories/meats.jpg"
          />
        </a>
      </span>
      <span
        className="categoryBox"
        data-testid="categoryBox"
      >
        <a
          href="/category/3"
          onClick={[Function]}
        >
          <span
            className="categoryLabel"
          />
          <span
            className="categoryLabelText"
          >
            bakery
          </span>
          <img
            alt="bakery"
            className="categoryImage"
            data-cy="category-bakery"
            data-testid="category-bakery"
            src="/static/img/categories/bakery.jpg"
          />
        </a>
      </span>
      <span
        className="categoryBox"
        data-testid="categoryBox"
      >
        <a
          href="/category/4"
          onClick={[Function]}
        >
          <span
            className="categoryLabel"
          />
          <span
            className="categoryLabelText"
          >
            fruit & veg
          </span>
          <img
            alt="fruit & veg"
            className="categoryImage"
            data-cy="category-fruit & veg"
            data-testid="category-fruit & veg"
            src="/static/img/categories/fruit & veg.jpg"
          />
        </a>
      </span>
      <span
        className="categoryBox"
        data-testid="categoryBox"
      >
        <a
          href="/category/5"
          onClick={[Function]}
        >
          <span
            className="categoryLabel"
          />
          <span
            className="categoryLabelText"
          >
            cereals
          </span>
          <img
            alt="cereals"
            className="categoryImage"
            data-cy="category-cereals"
            data-testid="category-cereals"
            src="/static/img/categories/cereals.jpg"
          />
        </a>
      </span>
      <span
        className="categoryBox"
        data-testid="categoryBox"
      >
        <a
          href="/category/6"
          onClick={[Function]}
        >
          <span
            className="categoryLabel"
          />
          <span
            className="categoryLabelText"
          >
            drinks
          </span>
          <img
            alt="drinks"
            className="categoryImage"
            data-cy="category-drinks"
            data-testid="category-drinks"
            src="/static/img/categories/drinks.jpg"
          />
        </a>
      </span>
    </div>
  </div>
  <div
    className="sc-AxhUy rbvJG"
  >
    <br />
    <span
      className="sc-AxgMl jzkpql"
    >
      <a
        data-cy="privacy"
        data-testid="privacy"
        href="/privacy"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Privacy
        </button>
      </a>
      <a
        data-cy="contact"
        data-testid="contact"
        href="/contact"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Contact
        </button>
      </a>
       © 2019 the affable bean company
    </span>
  </div>
</div>
`;
